ring/partition_ring_watcher.go

```diff
@@ -91,9 +91,9 @@ func (w *PartitionRingWatcher) updatePartitionRing(desc *PartitionRingDesc) {
 	}
 }
 
-// GetRing returns the most updated snapshot of the PartitionRing. The returned instance
+// PartitionRing returns the most updated snapshot of the PartitionRing. The returned instance
 // is immutable and will not be updated if new changes are done to the ring.
-func (w *PartitionRingWatcher) GetRing() *PartitionRing {
+func (w *PartitionRingWatcher) PartitionRing() *PartitionRing {
```

Comment:
Note to reviewers: renamed to adhere to the `PartitionRingReader` interface.

Reply:
That's fine, but it can return `nil` value if watcher is not running yet. We should avoid that.

Reply:
Good job spotting it! I haven't realised it, and was unintentional. Fixed in https://github.com/grafana/dskit/pull/479/commits/6fcc58211fbb3e843f990eab2de285586e13bb43